src/service/xhr-impl.js

```diff
@@ -223,7 +223,7 @@ function setupJson_(init) {
   init.headers = init.headers || {};
   init.headers['Accept'] = 'application/json';
 
-  if (init.method == 'POST') {
+  if (init.method == 'POST' && init.type == 'json') {
```

Comment:
Still need to work on this to make sure I didn't break anything, but I was trying to make fetchJson more allow sending `FormData` type instead of only JSON - fetchJson would still return JSON though. 

Maybe it'd be better to create a different method for that, `postData` or something.